### DISCUSSION TITLE
"raise" isn't supported within entry actions

### DISCUSSION THREAD

**Discussion by Bab64** [statelyai/xstate#4326] [Q&A] [answered]
I'm the author of the "Reactive Internet Programming-SCXML in action" book and I've tried X-STATE to implement some of the case studies in my book. X-STATE is amazing! Congratulations...
I've found an issue, not really a bug (code below). It's common in modeling to communicate **WITHIN the same machine** between parallel states. While 'raise' in SCXML/X-STATE is the tool for, it seems that we cannot in X-STATE send events within entry actions? In my opinion, there is no difficulty to queue such event... Looking at my code below, "\tEntering 'B'... again?" has to be displayed **TWICE** while, as this time, it's displayed only one time. REGARDS.

const X = X_STATE.createMachine({
    id: 'X',
    initial: 'A',
    type: "parallel",
    states: {
        A: { // Sending events within 'entry' fails:
            entry: () => X_STATE.raise('e2')
        },
        B: {
            on: {
                e2: {target: 'B'}
            },
            entry: () => console.info("\tEntering 'B'... again?")
        }
    }
});
(X_STATE.interpret(X).onTransition((state) => {
    console.log(state.value);
})).start();

**Comment by davidkpiano** [comment, marked as answer]
Hey, you're close! `raise(...)` is an **action creator**; it doesn't imperatively execute the action.

So instead, the entry action should look like:

```ts
// creates an action object like { type: 'xstate.raise', event: { type: 'e2 } }
entry: raise({ type: 'e2' })
```

**Reply by Bab64** [reply]
Thanks! this works, but solution code is a bit "non-intuitive"...
Guess that the doc. (https://xstate.js.org/docs/guides/actions.html#raise-action) may be more clear since the above solution is not discussed?
Regards.
